nwcsafpps_runner/l1c_processing.py

```diff
@@ -142,10 +144,24 @@ def run(self, msg):
         l1c_proc = LVL1C_PROCESSOR_MAPPING.get(self.service)
         if not l1c_proc:
             raise AttributeError("Could not find suitable level-1c processor! Service = %s" % self.service)
-
-        self.l1c_result = self.pool.apply_async(l1c_proc, (self.level1_files,
-                                                           self.result_home),
-                                                self._l1c_processor_call_kwargs)
+        LOG.debug(
+            "Starting level1c processing in separate process.")
+        manager = Manager()
+        result_dict = manager.dict()
+        process1 = Process(
+            name="Process one level1c file in a separate process.",
+            target=self.run_inner,
+            args=(l1c_proc, result_dict))
+        process1.start()
+        process1.join(self.time_limit_seconds)  # Normally takes 3-8s for VIIRS
+        self.l1cfile = result_dict.get("l1cfile", None)
+        try:
+            process1.close()
+        except ValueError:
+            # if process1.is_alive():
+            LOG.warning(
+                "Processing level1c file not terminated after 60s.")
+            process1.terminate()
```

Comment:
So, I'm not sure I follow this. But I think the idea is to start each new processing in a separate process. Without any limitation than processing time. I think that quickly can get you into problems if your processing is using more memory than you have available.

Reply:
It starts a new process but also waits for it to finish (with join). So there is never more than one active process. The limit is just to clean up if a process hangs for any reason. In the end I made the time limit configurable with 60s as default. (I will fix the message.)